### DISCUSSION TITLE
What's the happy medium between using TailwindCSS colours and maintaing brand/design system colours?

### DISCUSSION THREAD

**Discussion by studiovoyager** [tailwindlabs/tailwindcss#10841] [Help]
I have been working on this problem over a few projects the last six months - how do I work out a compromise or happy medium between tailwind CSS colour eg. bg-red-700 vs brand/design system approaches of brand-primary, brand-secondary etc.

Here are a few cases I'm dealing with:

A site that requires multiple configs for accessibility.

eg. 
Default colours
Dark mode colours
Colour blind colours
Colour blind dark mode colours

If I use vanilla tailwind inside all our components, I need to redefine those per 'theme' eg.
`bg-red-700` for a button must be redefined for colour blind ie. `bg-red-700: 'blue'`

I've been using tailwindcss themer (https://github.com/RyanClementsHax/tailwindcss-themer) to switch themes, which is working well, but have had to use brand-primary, brand-secondary etc. and redefine these colour sets per theme.

Another case is building a boilerplate for multiple sites where you can switch our a tailwind config to style:

Here I can apply some vanilla tailwind css classes across all my components, and then for a new site just do a search+replace to change all the 'red' for 'blue' if thats what the site brand is. Failing that, I would have to re-define those colours as something else ie. red is actually blue `bg-red-700: 'blue'` which just don't seem right :)

I think what is helpful with thinking through this is recognising that when colour is used (perhaps specifically, when its used by a designer) it has a meaning behind it. ie using red on the site, means we are defining a primary action, or its a key brand colour, its not just a colour. So perhaps, the design system is defined somewhere with very specific meaning and value tied to colours (rules) eg. Figma, but then in the code its simply a colour 'bg-red-700'

Using things like 'brand-primary-700' is bring the colour rules right into the code? I might be starting to ramble now :)

Anyway, I'd like to discuss this.

**Comment by studiovoyager** [comment]
I wonder if a few more variants could help with the problem. ie.
`
:dark
:grayscale
:deuteranopia (colour blind)
:tritanopia (colour blind)
`
ie. Inside a button component
`
Button(class="bg-red-700 dark:bg-red-200  grayscale:bg-neutral-600 deuteranopia:bg-teal-600 ...
`

**Reply by wongjn** [reply]
Yes, this would be the approach I'd take.
